webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/users/RudderUserDetailsFile.scala

```diff
@@ -76,34 +75,18 @@ final case class UserFile(
     inputStream: () => InputStream
 )
 
+sealed abstract class PasswordEncoder(override val entryName: String) extends EnumEntry {
+  def name:        String = entryName
+  def displayName: String = name
+  def encoder:     PasswordEncoder.Rudder
+}
+
 //Password encoder type definition. Done like that to avoid
 //a deprecation warning on `PasswordEncoder` thanks to @silent annotation
-object PasswordEncoder {
```

Comment:
For all that, get a look to what @amousset is doing in https://github.com/Normation/rudder/pull/5604

I think it's compatible with it, but I don't how hard the other subject will become.